kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/ZooKeeperClientProvider.scala

```diff
@@ -71,6 +73,27 @@ object ZooKeeperClientProvider extends Logging {
     builder.build()
   }
 
+  def getGracefulStopThreadDelay(conf: KyuubiConf): Long = {
+    val baseSleepTime = conf.get(HA_ZK_CONN_BASE_RETRY_WAIT)
+    val maxSleepTime = conf.get(HA_ZK_CONN_MAX_RETRY_WAIT)
+    val maxRetries = conf.get(HA_ZK_CONN_MAX_RETRIES)
+    val retryPolicyName = conf.get(HA_ZK_CONN_RETRY_POLICY)
+    RetryPolicies.withName(retryPolicyName) match {
+      case ONE_TIME => baseSleepTime
+      case N_TIME => maxRetries * baseSleepTime
+      case BOUNDED_EXPONENTIAL_BACKOFF =>
+        (0 until maxRetries).map { retryCount =>
+          val retryWait = baseSleepTime * Math.max(1, Random.nextInt(1 << (retryCount + 1)))
+          Math.min(retryWait, maxSleepTime)
+        }.sum
```

Comment:
shall we compare this value and `.min(maxSleepTime * maxRetries)` ?

Reply:
not necessary